app.js

```diff
@@ -48,10 +37,38 @@ angular.module('PathOfDamage', [])
         }
       }
     }
-    $scope.updateHits(true);
-    if (!skipStringify) {
-      $scope.stringifyUrlData();
+    if (!skipUpdates) {
+      $scope.updateHits();
+    }
+  };
+
+  $scope.updateHits = function (skipSerialization) {
+    for (var i = 0; i < $scope.hits.length - 1; i++) {
+      $scope.hits[i] = calcDamage($scope.hits[i].hit);
+    }
+    if (!skipSerialization) {
+      serializeData();
+    }
+  };
+
+  $scope.updateHitTable = function (index) {
+    if ($scope.hits[index].hit === null) {
```

Comment:
Do we care about null specifically here? can we just use
if (!$scope.hits[index].hit)